### ISSUE TITLE
On Notification Event?

### ISSUE THREAD

**Issue by agarrharr** [avivais/phonegap-parse-plugin#8]
I was able to easily get push notifications on my device with this plugin, but is there a way to get the message from the push notification in the app? In other words, is there an event that is triggered when a notification is received or when the app is opened from a notification?

For example: if I send the notification that says "Hello", it will show up in the notification drawer and it will say "Hello", but when I open it, the app does nothing. I would like to show a popup alert with the message "Hello" when the app is opened via the notification.

**Comment by mypark** [comment]
+1 for this as well

**Comment by thomasMary** [comment]
+1

**Comment by mypark** [comment]
I noticed that there are a few forks that implement the on notification, 

https://github.com/taivo/parse-push-plugin
https://github.com/nickh364/phonegap-parse-plugin

and also a few that have updated only the ios, or android side of it respectively

https://github.com/cookys/parse-push-plugin
https://github.com/4ndywilliamson/Parse-Push-Plugin-Android

Not sure which ones are more stable, the docs have not been updated on most of them. 

Does anyone have a suggestion on which ones are more stable?

**Comment by Montoya** [comment]
@mypark 

I tried taivo and it caused some errors, didn't work at all. 

Tried cookys and it works really nicely with Android so far, registers perfectly and lets me display notifications in the app. Gonna keep testing on iOS.

**Comment by Montoya** [comment]
OK let me correct myself. cookys works really well on Android except for 1 issue: when opening the app from a push notification, the app hangs. I can't pinpoint the issue but I'll try to figure it out tomorrow :(

**Comment by Montoya** [comment]
OK wait. cookys has one issue I can't resolve: opening the app from a push notification on android causes the app to hang (there's a javascript error but I can't pinpoint it). So frustrating.

**Comment by justinwright0** [comment]
@Montoya If there is some message about 'You must call Parse.initialize first' then I would suggest you check out this implementation.
https://github.com/jarseneault/phonegap-parse-plugin/tree/ldr

**Comment by Montoya** [comment]
Well I went back to this repo (avivais) and I'm in the same place now, notifications don't crash or hang the app but nothing is passed to JS. Which is fine for now I guess. I realize most apps don't display push notifications in the app, but it would have been nice to be able to get custom data from push notifications. Anyway, there's got to be a hundred repos of this plugin that all seem to do something slightly different but the documentation is either not updated at all or poorly written, so it's really a mess to go through them. I'm giving up before I lose any more sleep =/

**Comment by mypark** [comment]
@Montoya you can check out my fork of cooksys, where I fixed a few of the bugs you mentioned... https://github.com/mypark/parse-push-plugin

I also backported to the deprecated initialize instead of pluginInitialize since I'm using ionic instead of the latest cordova.

**Comment by Montoya** [comment]
Dang I wish I had found your fork sooner! Will try it soon, thanks!

**Comment by mypark** [comment]
haven't updated the docs and haven't been testing/maintaining it but there are somethings that you have to do in the code itself, like Parse.initialize - so I'm not sure if that's adequately stated in the docs ... anyways, at least some of the commits can save you some time in the errors I ran into

**Comment by sunshineo** [comment]
https://github.com/mypark/parse-push-plugin forked from
https://github.com/cookys/parse-push-plugin forked from
https://github.com/taivo/parse-push-plugin forked from
https://github.com/benjie/phonegap-parse-plugin forked from
https://github.com/hyliker/phonegap-parse-plugin forked from
https://github.com/xydudu/phonegap-parse-plugin forked from
https://github.com/avivais/phonegap-parse-plugin (which is this repository)

Why? What is going on? What should I use?? Can some one merge fixes all the way back here?
Thanks!

**Comment by avivais** [comment]
I merge all the open PRs

Avi Vaisenberger
Co-Founder & CTO
Fabula Gaming <http://www.fabulagame.com>
+972-50-766-6550

**Comment by Montoya** [comment]
@sunshineo all the implementations are very different at this point. I would recommend trying mypark's fork, otherwise stick with this one. most of the other ones are broken in one way or another.

**Comment by Montoya** [comment]
Hey @mypark , could you help me? What's the function that I use in Javascript to receive the notification when the app is running and display it? The documentation says "onNotification(pnObj)" but that's not working for me.

**Comment by Montoya** [comment]
OK @mypark help me understand here... your plugin lets push notifications deep link into the app by implementing openURL but it removes the functionality that allows basic text push notifications to be displayed in the app... ? So if a push is received while the app is running, nothing happens...?

**Comment by Montoya** [comment]
Also I tried deep link to an external URL and nothing happened... can I only deep-link within the app?

**Comment by Montoya** [comment]
Also I probably sound like a broken record but I was looking at some push notification plugins outside the Parse sphere and quite a few support display notifications while the app is in the foreground: 

https://github.com/AppGyver/PushNotifications

https://github.com/phonegap-build/PushPlugin

I've looked at a bunch of forks of this one and I can't tell if any of them do it. A couple seemed to but had various bugs preventing it from actually working. And the rest don't have updated documentation so it's really hard to tell what they do and how to use it. 

So, here's what I'm wondering: 
- Can these READMEs get updated with code examples? 
- If a fork has diverged from this master repo, could it be turned into a new separate repo so we can have issue threads over there? 
- Could someone do a tutorial on how to display notifications while the app is running, for iOS and Android, if you already have it working in your app?

**Comment by grrrian** [comment]
I added a Pull Request for notification callbacks: https://github.com/avivais/phonegap-parse-plugin/pull/52. In the mean time it is available in my branch: https://github.com/grrrian/phonegap-parse-plugin

**Comment by Montoya** [comment]
Sounds very nice and I promise to check it out soon. Don't forget to update README.md btw! Or not! It's cool :)

**Comment by grrrian** [comment]
Credits to mypark for the inspiration.

**Comment by mypark** [comment]
@Montoya sorry I didn't write back earlier - been out of the loop on this for quite a while.  Did you ever get a plugin that worked for you?

**Comment by Montoya** [comment]
No, but I also haven't cared since I heard that Parse is shutting down -__-
